src/main/kotlin/gg/essential/elementaPractice/gui/components/ColorToolComponent.kt

```diff
@@ -0,0 +1,59 @@
+package gg.essential.elementaPractice.gui.components
+
+import gg.essential.elementa.components.UIImage
+import gg.essential.elementa.constraints.ImageAspectConstraint
+import gg.essential.elementa.constraints.PixelConstraint
+import gg.essential.elementa.constraints.animation.Animations
+import gg.essential.elementa.dsl.animate
+import gg.essential.elementa.dsl.constrain
+import gg.essential.elementa.dsl.effect
+import gg.essential.elementa.dsl.pixels
+import gg.essential.elementa.effects.OutlineEffect
+import gg.essential.elementaPractice.gui.WhiteboardGUI
+import java.awt.Color
+import java.util.concurrent.CompletableFuture
+import javax.imageio.ImageIO
+
+public class ColorToolComponent constructor(
+    private val colorTool: WhiteboardGUI.ColorTools
+) : UIImage(CompletableFuture.supplyAsync {
+    ImageIO.read(this::class.java.getResourceAsStream("/assets/elementapractice/${colorTool.fileName}"))
+}) {
+
+    public val outline: OutlineEffect =
+        OutlineEffect(color = Color.WHITE, width = 1f, sides = setOf(OutlineEffect.Side.Bottom))
+
+    public fun getToolColor(): Color = colorTool.color
+
+    init {
+
+        constrain {
+            width = 30.pixels()
+            height = ImageAspectConstraint()
+        }
+
+        onMouseClick {
+            setWidth(32.pixels())
```

Comment:
This is likely redundant because in order for the component to be clicked, the hover event will be fired and the width will already be set to 32